app/models/optOut/OptOutUpdateRequestModel.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.optOut
+
+import play.api.libs.json.{Format, Json}
+import play.mvc.Http.Status.NO_CONTENT
+
+object OptOutUpdateRequestModel {
+
+  private val defaultUpdateReason: Int = 10
+  private val successfulStatusCode: Int = NO_CONTENT
+
+  case class OptOutUpdateRequest(taxYear: String, updateReason: Int = defaultUpdateReason)
+  sealed trait OptOutUpdateResponse {
+    val statusCode: Int
+  }
+  case class OptOutUpdateResponseSuccess(correlationId: String, statusCode: Int = successfulStatusCode) extends OptOutUpdateResponse
+  case class ErrorItem(code: String, reason: String)
+  case class OptOutUpdateResponseFailure(statusCode: Int, failures: List[ErrorItem]) extends OptOutUpdateResponse
```

Comment:
I'm not really a fan of using polymorphic response objects like this. I think we might be more idiomatic with an Either or Future. However, I know this is a pattern in the app. Maybe one to bring up at Tech Deep Dive sometime?

Reply:
I guess we still need the OptOutUpdateResponseFailure to parse the response but I can/prefer Either[ErrorCode, OptOutUpdateResponseSuccess], happy to change, shall I?

Reply:
We have to map response in the way @james-king-hmrc  implemented this, its a standard approach across FE and BE, so should be fine, Future[Either[_,_]] type is used in the up stream logic close to the core domain

Reply:
yes indeed i was gonna keep OptOutUpdateResponseFailure but return Either[ErrorCode, OptOutUpdateResponseSuccess]

Reply:
Yeah. I know this is a standard pattern, but it still feels less than optimal. I'm happy to stick with it for now, but maybe this is one to discuss in a Tech Deep Dive? Feels like we should be moving away from this?

Reply:
i think returning Either helps in exiting (when in error) the monad bind loop in a more standard way without the extra step of checking if the response was a failure or not